samples/matter/common/src/binding_handler.cpp

```diff
@@ -5,83 +5,82 @@
  */
 
 #include "binding_handler.h"
-#include "temp_sensor_manager.h"
-#include "temperature_measurement/sensor.h"
-
-#include <app-common/zap-generated/ids/Clusters.h>
-#include <app-common/zap-generated/ids/Commands.h>
-#include <app/CommandSender.h>
-#include <app/OperationalSessionSetup.h>
-#include <app/clusters/bindings/BindingManager.h>
-#include <app/clusters/bindings/bindings.h>
-#include <app/data-model/Decode.h>
-#include <app/util/binding-table.h>
-#include <controller/InvokeInteraction.h>
-#include <controller/ReadInteraction.h>
-#include <lib/support/CodeUtils.h>
+
+#include <zephyr/logging/log.h>
 
 LOG_MODULE_DECLARE(app, CONFIG_CHIP_APP_LOG_LEVEL);
 
 using namespace chip;
 using namespace chip::app;
 
-void BindingHandler::Init()
+void BindingHandler::Init(deviceChangedHandler pHandler)
```

Comment:
Could I use `VerifyOrReturn(mDeviceChangedHandler != nullptr, LOG_ERR("Invalid device handler");` as in the other methods? Or it is better to return CHIP_ERROR?

Reply:
In my opinion it would be better to return an error. Otherwise the caller actually doesn't know that initialization failed and the module is not going to work properly.